src/copy.jl

```diff
@@ -45,3 +45,100 @@ function defaultcopy!(dest::MOI.AbstractInstance, src::MOI.AbstractInstance)
 
     idxmap
 end
+
+"""
+    allocatevariables!(instance::MOI.AbstractInstance, nvars::Integer)
+
+Creates `nvars` variables and returns a vector of `nvars` variable indices.
+"""
+function allocatevariables! end
+
+"""
+    allocateobjective!(instance::MOI.AbstractInstance, sense::MOI.OptimizationSense, f::MOI.AbstractFunction)
+
+Informs `instance` that `loadobjective!` will be called with the same arguments after `loadvariables!` is called.
+"""
+function allocateobjective! end
```

Comment:
Objective should not be a special case. This should work for all attributes.

Reply:
Should `MOIU.defaultcopy!` also copy more attributes ?

Reply:
Yes, we shouldn't have attributes unexpectedly lost during `copy!`. We may need a `ListOfCopyableAttributes`...

Reply:
It also seems like variable and constraint attributes need to be provided to `allocatevariables` and `allocateconstraint`. It's possible that an attribute could affect transformations like scaling, or attributes could be used on a constraint-wise basis to choose which transformation to apply.

Reply:
It would be a bit tricky to add the list of attribues to allocatevariables and allocateconstraints. We could have `allocateset!` and `loadset!` and use it for every attribute. One other option would be to use `set!` after `loadvariables!`. What do you think ?

Reply:
The attributes (like scaling) might be needed to decide which transformations to perform. How would that still work in the one-pass framework?